src/urllib3/_collections.py

```diff
@@ -432,3 +432,34 @@ def _has_value_for_header(self, header_name: str, potential_value: str) -> bool:
         if header_name in self:
             return potential_value in self._container[header_name.lower()][1:]
         return False
+
+    def __ior__(self, other: object) -> HTTPHeaderDict:
+        # Supports extending a header dict in-place using operator |=
+        # combining items with add instead of __setitem__
+        maybe_constructable = ensure_can_construct_http_header_dict(other)
+        if maybe_constructable is None:
+            return NotImplemented
+        other_as_http_header_dict = type(self)(maybe_constructable)
```

Comment:
Are there any advantages of the conversion in `__ior__` and `__or__`?

Reply:
IIRC mypy was unhappy about that, because `__ior__` and `__or__` must accept any object as "other" (and will opt-out of the comparison by returning `NotImplemented`), but `HTTPHeaderDict.extend` explicitly wants a `ValidHTTPHeaderSource`.

Reply:
This creates new objects and looks to have little functional benefit because `extend` will do similar kind of validation. Can `typing.cast` be used instead?

Reply:
Yes, likely. I'll try it out when I have the spare time and report back.

Reply:
Nice, thanks!